src/ngraph/frontend/onnx_import/op/slice.cpp

```diff
@@ -36,6 +38,99 @@ namespace ngraph
     {
         namespace op
         {
+            namespace set_10
+            {
+                namespace
+                {
+                    std::vector<int64_t> axes_to_mask(const std::vector<int64_t>& axes)
+                    {
+                        std::vector<int64_t> mask(
+                            *std::max_element(std::begin(axes), std::end(axes)) + 1, 1);
+                        for (int i = 0; i < axes.size(); ++i)
+                        {
+                            mask[axes[i]] = 0;
+                        }
+                        return mask;
+                    }
+                }
+
+                NodeVector slice(const Node& node)
+                {
+                    NodeVector inputs{node.get_ng_inputs()};
+                    const auto data = inputs.at(0);
+
+                    auto starts = inputs.at(1);
+                    auto ends = inputs.at(2);
+
+                    // Slice is calculated over all axes as default
+                    std::shared_ptr<ngraph::Node> axes;
+                    if (inputs.size() >= 4) // axes input provided
+                    {
+                        axes = inputs.at(3);
+                        NGRAPH_CHECK(axes->is_constant(), "Axes input must be constant");
+                    }
+                    else
+                    {
+                        const size_t data_rank =
+                            data->get_output_partial_shape(0).rank().get_length();
+                        axes = default_opset::Constant::create(
+                            element::i64,
+                            {data_rank},
+                            common::get_monotonic_range<int64_t>(data_rank));
+                    }
+
+                    const bool is_steps_provided = inputs.size() == 5;
+                    std::shared_ptr<ngraph::Node> steps = nullptr;
+                    if (is_steps_provided)
+                    {
+                        steps = inputs.at(4);
+                    }
+
+                    const auto axes_const = as_type_ptr<default_opset::Constant>(axes);
+                    const auto axes_vec = axes_const->cast_vector<int64_t>();
+
+                    const auto data_rank = data->get_output_partial_shape(0).rank();
+                    NGRAPH_CHECK(data_rank.is_static(), "Data rank of input must be static");
+                    NGRAPH_CHECK(data_rank.get_length() == axes_vec.size(),
```

Comment:
Why axes size should be equal to rank size? axes input can have any positive size <= data_rank.get_length().

Reply:
@GlebKazantaev  In PR to master branch https://github.com/NervanaSystems/ngraph/pull/4470 such limitations do not exist but it requires some changes to `StridedSlice` operator which are not allowed on release branch:
1.  default `strides` length is calculated in incorrect way
2. cases such as: we have 4D input data, 2-elements `begins` and `ends` and we want to apply it to 2, 3 axes are not supported
In order to provide version which support some expected model such limitations are acceptable but generally you are right